src/editor/components/StylesColor.js

```diff
@@ -34,12 +36,56 @@ const Color = ( { selector } ) => {
 		setUserConfig( config );
 	};
 
+	/**
+	 * Define mouse state and related functions
+	 */
+	const [ isMouseDown, setIsMouseDown ] = useState( false );
+	const handleMouseDown = () => setIsMouseDown( true );
+	const handleMouseUp = () => setIsMouseDown( false );
+
+	/**
+	 * Define colour variables, used to avoid jumping colour picker when ContrastChecker display toggles
+	 */
+	const [ textColour, setTextColour ] = useState( colorStyles.text );
+	const [ backgroundColour, setBackgroundColour ] = useState(
+		colorStyles.background
+	);
+
+	/**
+	 * When the values are changed, ensure the mouse buttons have been released before updating
+	 */
+	useEffect( () => {
+		if ( ! isMouseDown ) {
+			setTextColour( colorStyles.text );
+			setBackgroundColour( colorStyles.background );
+		}
+	}, [ isMouseDown, colorStyles.text, colorStyles.background ] );
+
+	/**
+	 * Add and remove the mouse event listeners
+	 */
+	useEffect( () => {
+		const container = containerRef.current;
+
+		if ( container ) {
+			container.addEventListener( 'mousedown', handleMouseDown );
+			container.addEventListener( 'mouseup', handleMouseUp );
+		}
+
+		return () => {
+			if ( container ) {
+				container.removeEventListener( 'mousedown', handleMouseDown );
+				container.removeEventListener( 'mouseup', handleMouseUp );
+			}
+		};
+	}, [] );
```

Comment:
Just an FYI as I've advised against this approach in another comment, but in React, we can access any events via props. So just as we use `onClick` often, it's possible to add `onMouseDown` and `onMouseUp` directly to the `div` element that we're targeting here, which reduces this complexity quite a bit.